src/Web64/Nlp/NlpClient.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\strawberry_runners\Web64\Nlp;
+
+use Web64\Nlp\NlpClient as Web64NlpClient;
+
+class NlpClient extends Web64NlpClient {
+
+  const MAX_RETRY_HOST = 3;
+
+  /**
+   *  Get language code for text
+   *
+   * @param     $text
+   * @param int $predictions
+   *
+   * @return mixed|null
+   */
+  public function fasttext($text, $predictions = 1) {
+    //Predictions at least 1 are required if POSTING. GET is more forgiving
+    //Also. We need to URL encode upfront here.
+    $data = $this->post_call(
+      '/fasttext', ['text' => rawurlencode($text), 'predictions' => $predictions]
+    );
+    if (isset($data['fasttext']) && isset($data['fasttext']['language'])) {
+      return $data['fasttext'];
+    }
+    return NULL;
+  }
+
+  public function get_call($path, $params, $retry = 0) {
+    $url = $this->api_url . $path;
+
+    $retry++;
+
+    if (!empty($params)) {
+      $url .= '?' . http_build_query($params);
+    }
+
+    $result = @file_get_contents($url, FALSE);
+
+    $response = isset($http_response_header) && isset($http_response_header[0])
+      ? $this->isValidResponse($http_response_header[0]) : FALSE;
+    if ($response && !empty($result)) {
+      return json_decode($result, 1);
+    }
+    else {
+      if ($retry >= static::MAX_RETRY_HOST) {
+        return NULL;
+      }
+      $this->chooseHost();
+      return $this->post_call($path, $params, $retry);
+    }
+  }
+
+  public function post_call($path, $params, $retry = 0) {
+    $url = $this->api_url . $path;
+    $retry++;
+    if ($retry > static::MAX_RETRY_HOST) {
+      return NULL;
+    }
+    $opts = [
+      'http' =>
+        [
+          'method'  => 'POST',
+          'header'  => 'Content-type: application/x-www-form-urlencoded',
+          'content' => http_build_query($params),
+        ],
+    ];
+    $context = stream_context_create($opts);
+    $result = @file_get_contents($url, FALSE, $context);
```

Comment:
Is this how the NLPServer's PHP library was interacting with the server?!?!?? I only ever looked at the server-side code and never at the client. Wow. I had assumed it was using GET and the form-based parsing was just for the crappy web "interface" it ships with.

I will definitely rewrite this as a pure JSON interface and hopefully then we can get rid of all of this gnarly code :grimacing: